Neo4j.Driver/Neo4j.Driver/Internal/IO/MessageReader.cs

```diff
@@ -98,4 +107,16 @@ private void ProcessMessage(IResponsePipeline pipeline, PackStreamReader packStr
             throw new ProtocolException($"Unknown response message type {message.GetType().FullName}");
         }
     }
+
+    public async ValueTask DisposeAsync()
+    {
+        if (BufferStream is IAsyncDisposable bufferStreamAsyncDisposable)
```

Comment:
Does MemoryStream implement IAsyncDiposable or not? Seems like an unnecessary conditional?

Reply:
this is a blunder, will fix